renku/core/commands/dataset.py

```diff
@@ -158,18 +159,25 @@ def add_to_dataset(
     name,
     link=False,
     force=False,
+    create=False,
     sources=(),
     destination='',
     with_metadata=None,
-    urlscontext=contextlib.nullcontext
+    urlscontext=contextlib.nullcontext,
+    fail_if_exists=True
```

Comment:
I think this should be the default behavior. Is there a reason for this flag? I see problems with `force` and `fail_if_exists` cause it can lead to undefined behavior since we don't have explicitly defined precedence anywhere.

Reply:
In some other commands (e.g. `renku dataset create`) we need to be able to re-create a dataset (`create` prompts the user itself) so we need this flag. `force` is to tell git to force-add ignored files; dataset duplication tests is performed before and if it fails we don't reach the add part. Would you please explain how they can cause problem?

Reply:
should the files be removed from the dataset if it is recreated? This doesn't seem to be the case atm

Reply:
Good catch! Should we remove this ability of recreating a dataset? We can simply instruct user to remove the dataset first if they want to recreate it. This would simplify the code in some places.

Reply:
I think so - the behavior is also not consistent between using the `--create` flag and using `renku dataset create` - in the former you get an error, in the latter you are asked if you want to overwrite

Reply:
I'll create an issue for it.

Reply:
I think `renku dataset create` should also just create an error and it would be fine for the time being